src/idl/src/parser.y

```diff
@@ -0,0 +1,1126 @@
+      { TRY(idl_create_declarator(pstate, LOC(@1.first, @2.last), $1, $2, &$$)); }
+  ;
+
+fixed_array_sizes:
+    fixed_array_size
+      { $$ = $1; }
+  | fixed_array_sizes fixed_array_size
+      { $$ = idl_push_node($1, $2); }
+  ;
+
+fixed_array_size:
+    '[' positive_int_const ']'
+      { $$ = $2; }
+  ;
+
+simple_declarator:
+    identifier
+      { TRY(idl_create_declarator(pstate, &@1, $1, NULL, &$$)); }
+  ;
+
+complex_declarator: array_declarator ;
+
+typedef_dcl:
+    "typedef" type_spec declarators
+      { TRY(idl_create_typedef(pstate, LOC(@1.first, @3.last), $2, $3, &$$)); }
+  ;
+
+declarators:
+    declarator
+      { $$ = $1; }
+  | declarators ',' declarator
+      { $$ = idl_push_node($1, $3); }
+  ;
+
+declarator:
+    simple_declarator
+  | complex_declarator
+  ;
+
+identifier:
+    IDL_TOKEN_IDENTIFIER
+      { $$ = NULL;
+        size_t n;
+        bool nocase = (pstate->flags & IDL_FLAG_CASE_SENSITIVE) == 0;
+        if (pstate->parser.state == IDL_PARSE_ANNOTATION_APPL)
+          n = 0;
+        else if (pstate->parser.state == IDL_PARSE_ANNOTATION)
+          n = 0;
+        else if (!(n = ($1[0] == '_')) && idl_iskeyword(pstate, $1, nocase))
+          SEMANTIC_ERROR(pstate, &@1, "Identifier '%s' collides with a keyword", $1);
+
+        if (pstate->parser.state != IDL_PARSE_UNKNOWN_ANNOTATION_APPL_PARAMS)
+          TRY(idl_create_name(pstate, &@1, idl_strdup($1+n), &$$));
+      }
+  ;
+
+annotation_dcl:
+    annotation_header '{' annotation_body '}'
+      { $$ = NULL;
+        /* discard annotation in case of redefinition */
+        if (pstate->parser.state != IDL_PARSE_EXISTING_ANNOTATION_BODY)
+          $$ = $1;
+        TRY(idl_finalize_annotation(pstate, LOC(@1.first, @4.last), $1, $3));
+      }
+  ;
+
+annotation_header:
+    "@" "annotation"
+      { pstate->annotations = true; /* register annotation occurence */
+        pstate->parser.state = IDL_PARSE_ANNOTATION;
+      }
+    identifier
+      { TRY(idl_create_annotation(pstate, LOC(@1.first, @2.last), $4, &$$)); }
+  ;
+
+annotation_body:
+    %empty
+      { $$ = NULL; }
+  | annotation_body annotation_member ';'
+      { $$ = idl_push_node($1, $2); }
+  | annotation_body enum_dcl ';'
+      { $$ = idl_push_node($1, $2); }
+  | annotation_body const_dcl ';'
+      { $$ = idl_push_node($1, $2); }
+  | annotation_body typedef_dcl ';'
+      { $$ = idl_push_node($1, $2); }
+  ;
+
+annotation_member:
+    annotation_member_type simple_declarator annotation_member_default
+      { TRY(idl_create_annotation_member(pstate, LOC(@1.first, @3.last), $1, $2, $3, &$$)); }
+  ;
+
+annotation_member_type:
+    const_type
+      { $$ = $1; }
+  | any_const_type
+      { $$ = $1; }
+  ;
+
+annotation_member_default:
+    %empty
+      { $$ = NULL; }
+  | IDL_TOKEN_DEFAULT const_expr
+      { $$ = $2; }
+  ;
+
+any_const_type:
+    IDL_TOKEN_ANY
+      { TRY(idl_create_base_type(pstate, &@1, IDL_ANY, &$$)); }
+  ;
+
+annotation_appls:
```

Comment:
The shift/reduce conflicts are all on "@" and originate here. Bison's shift/reduce conflict handling resolves them as shifts and that'll probably do the right thing here. That means a `%expect 9` could be an option, but it is better to eliminate the conflicts. I think (so you'd better check ...) that this can be done quite easily:

```
annotation_appls:
    %empty
      { $$ = NULL; }
  | nonempty_annotation_appls
      { $$ = $1; }
  ;

nonempty_annotation_appls:
    annotation_appl
      { $$ = $1; }
  | nonempty_annotation_appls annotation_appl
      { $$ = idl_push_node($1, $2); }
  ;
```

(Just removing the first case of `annotation_appl` only eliminates half of the conflicts.)